### PR TITLE
Add support for component using ForwardRef

### PR DESCRIPTION
**Pull request by amanmahajan7**
Using [`react-is`](https://github.com/facebook/react/tree/master/packages/react-is) to test valid elements and components. I tested a custom editor using `ForwardRef` and everything works as expected

Also check https://github.com/facebook/react/issues/12453

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,7 +33,8 @@
     "@babel/polyfill": "^7.2.5",
     "classnames": "^2.2.6",
     "keymirror": "^0.1.1",
-    "prop-types": "^15.7.2"
+    "prop-types": "^15.7.2",
+    "react-is": "^16.8.4"
```
Comment (MayhemYDG) [inline package.json]: This should be a dependency of rdg once we move common.

File: packages/react-data-grid/src/ColumnComparer.js [hunk 2]
```diff
@@ -1,11 +1,11 @@
-import { isValidElement } from 'react';
+import { isElement } from 'react-is';
 
 export function sameColumn(a, b) {
   let k;
 
   for (k in a) {
     if (a.hasOwnProperty(k)) {
-      if ((typeof a[k] === 'function' && typeof b[k] === 'function') || (isValidElement(a[k]) && isValidElement(b[k]))) {
+      if ((typeof a[k] === 'function' && typeof b[k] === 'function') || (isElement(a[k]) && isElement(b[k]))) {
```
Comment (MayhemYDG) [inline packages/react-data-grid/src/ColumnComparer.js]: Shouldn't these typeof checks be replaced with `isValidElementType`?

File: package.json (hunk 1 above)
Comment (amanmahajan7) [reply package.json]: That's correct. For now I added it here

File: packages/react-data-grid/src/ColumnComparer.js (hunk 2 above)
Comment (amanmahajan7) [reply packages/react-data-grid/src/ColumnComparer.js]: That makes sense. For this PR I just replaced `React.isValidElement` with `ReactIs.isElement` to be on  the safer side. I will revisit this once we clean up column metrics logic.

**Pull request merged**